File: multi-gammon-util/src/main/java/org/akoshterek/backgammon/data/TrainDataLoader.java
package org.akoshterek.backgammon.data;

import org.akoshterek.backgammon.Constants;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.zip.GZIPInputStream;

/**
 * @author Alex
 *         date 21.09.2015.
 */
public class TrainDataLoader {
    private static List<TrainEntry> loadData(final InputStream is) {
        try (BufferedReader reader = new BufferedReader(new InputStreamReader(is))) {
            List<TrainEntry> data = new ArrayList<>();
            String line;
            while ((line = reader.readLine()) != null) {
                String[] tokens = line.split("\\s");
                TrainEntry entry = new TrainEntry();
                entry.positionId_$eq(tokens[0]);
                for(int i = 0; i < Constants.NUM_OUTPUTS; i++) {
                    entry.reward()[i] = Double.parseDouble(tokens[i + 1]);
                }

                data.add(entry);
            }
            return data;
        } catch (final Exception e) {
            throw new RuntimeException(e);
        }
    }

    public static List<TrainEntry> loadGzipResourceData(final String resource) {
        try(GZIPInputStream is = new GZIPInputStream(TrainDataLoader.class.getResourceAsStream(resource))) {
            return loadData(is);
        } catch (final Exception e) {
            throw new RuntimeException(e);
        }
    }
}

File: multi-gammon-util/src/main/java/org/akoshterek/backgammon/data/TrainDataLoader.scala
package org.akoshterek.backgammon.data

import java.util.zip.GZIPInputStream

import org.akoshterek.backgammon.Constants
import resource.managed

import scala.io.Source

/**
  * @author Alex
  *         date 21.09.2015.
  */
object TrainDataLoader {
  def loadGzipResourceData(resource: String): List[TrainEntry] = {
    managed(Source.fromInputStream(new GZIPInputStream(classOf[TrainEntry].getResourceAsStream(resource))))
      .acquireAndGet(source => {
        var data: List[TrainEntry] = List()
        for (line <- source.getLines()) {
          val tokens: Array[String] = line.split("\\s")
          val reward = for (i <- 0 until Constants.NUM_OUTPUTS) yield tokens(i + 1).toDouble
          val entry: TrainEntry = new TrainEntry(tokens(0), reward.toArray)
          data = entry :: data
        }
        data
      })
  }
}